### PR TITLE
Show in-progress applications on My VA v2

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
This PR adds everything needed to show benefit applications that were saved in progress.

## Testing done
Local + new unit tests + new Cypress test

## Screenshots
<details>
  <summary>mobile screen</summary>
  
image

</details>

<details>
  <summary>tablet screen</summary>
  
image

</details>

<details>
  <summary>desktop screen</summary>
  
image

</details>

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/dashboard-2/components/Dashboard.jsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ const Dashboard = ({
             totalDisabilityRating={props.totalDisabilityRating}
           />
         )}
-        <div className="vads-l-grid-container">
+        <div className="vads-l-grid-container medium-screen:vads-u-padding--2 small-desktop-screen:vads-u-padding--0">
```
Comment (erikphansen) [inline src/applications/personalization/dashboard-2/components/Dashboard.jsx:86]: the grid container adds a lot of horizontal padding that we don't want at large screen sizes.

File: src/applications/personalization/dashboard-2/tests/e2e/dashboard-loa3.cypress.spec.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import disableFTUXModals from '~/platform/user/tests/disableFTUXModals';
-import { loa3User } from '@@profile/tests/fixtures/users/user.js';
+import loa3User from '@@profile/tests/fixtures/users/user.js';
```
Comment (erikphansen) [inline src/applications/personalization/dashboard-2/tests/e2e/dashboard-loa3.cypress.spec.js:2]: Oops. This was resolving to `undefined` before. Which is embarassing.

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx [hunk 3]
```diff
@@ -1,7 +1,68 @@
...
+    <div>
+      <h2>Apply for benefits</h2>
+      <h3 className="vads-u-font-size--h4 vads-u-font-family--sans vads-u-margin-bottom--2p5">
+        Applications in progress
+      </h3>
+      {savedForms.length > 0 && (
+        <div className="vads-l-grid-container vads-u-padding--0">
+          <div className="vads-l-row">
+            {savedForms.map(form => {
+              const formId = form.form;
```
Comment (sanlouise) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx:27]: Do we want to optional chain here? `form.form` etc.?

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  presentableFormId,
+}) => {
+  return (
+    <div
+      className="vads-u-display--flex vads-l-col--12 medium-screen:vads-l-col--6 small-desktop-screen:vads-l-col--4 medium-screen:vads-u-padding-right--3 vads-u-padding-bottom--3"
+      data-testid="application-in-progress"
+    >
+      <div className="vads-u-display--flex vads-u-width--full vads-u-flex-direction--column vads-u-justify-content--space-between vads-u-align-items--flex-start vads-u-background-color--gray-lightest vads-u-padding--2p5">
+        <div>
+          <div style={{ textTransform: 'uppercase' }}>{presentableFormId}</div>
```
Comment (sanlouise) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx:21]: I think we can use `vaos-u-text-transform--uppercase` here, it's used elsewhere in the codebase.

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+          <h4 className="vads-u-font-size--h3 vads-u-margin-top--0">
+            {formTitle}
+          </h4>
+          <div className="vads-u-display--flex">
+            <i
+              aria-hidden="true"
+              className={`fas fa-fw fa-exclamation-circle vads-u-margin-right--1 vads-u-margin-top--0p5`}
+            />
+            <div>
```
Comment (sanlouise) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx:30]: nit: can we make this `div` a Fragment?

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        >
+          Continue your application
+        </a>
+      </div>
+    </div>
+  );
+};
+
+ApplicationInProgress.propTypes = {
+  // The URL that lets the user continue their application
```
Comment (sanlouise) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx:52]: Thanks for adding comments in the propTypes! 💯

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx [hunk 7]
```diff
@@ -1,7 +1,68 @@
...
+              const formTitle = `Application for ${formTitles[formId]}`;
+              const presentableFormId = presentableFormIDs[formId];
+              const { lastUpdated, expiresAt } = form.metadata;
```
Comment (sanlouise) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx:30]: Optional chaining might be a safeguard here?

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx [hunk 8]
```diff
@@ -1,7 +1,68 @@
...
+};
+
+const mapStateToProps = state => {
+  const profileState = selectProfile(state);
+  const { savedForms } = profileState;
+  const verifiedSavedForms = savedForms
+    .filter(isSIPEnabledForm)
+    .sort(sipFormSorter);
+
+  return { savedForms: verifiedSavedForms };
```
Comment (sanlouise) [inline src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx:65]: Is it just me or does this read more complex that necessary? Do we want to rename `savedForms` here to `verifiedSavedForms`?  Not too opinionated, but something like:

```
const mapStateToProps = state => {
  const verifiedSavedForms = selectProfile(state)?.savedForms
    .filter(isSIPEnabledForm)
    .sort(sipFormSorter);

  return { verifiedSavedForms };
```

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx (hunk 4 above)
Comment (erikphansen) [reply src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx:21]: thank you! I was looking for that! 😆

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx (hunk 5 above)
Comment (erikphansen) [reply src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplicationInProgress.jsx:30]: nope :( it's needed to group the two `p` tags together. This `div` sits to the right of the `i` defined right before it.

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx (hunk 7 above)
Comment (erikphansen) [reply src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx:30]: 🤔 Well we know the `form` will be defined. So ideally we'd want to make sure `metadata` is defined before destructuring. Is there a simple syntax to do that? I'm pretty sure `form.metadata?` won't help? Nope trying that make node unhappy.

Comment (erikphansen) [reply src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx:30]: Oh! If all you want to do is prevent an uncaught error you could do `const { lastUpdated, expiresAt } = form.metadata || {};`

File: src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx (hunk 8 above)
Comment (erikphansen) [reply src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx:65]: Sure, this was copy/pasted from elsewhere :)

Comment (erikphansen) [reply src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx:65]: Looking at this now it makes me realize we should make sure that the `savedForms` are an array before filtering...

**Review by sanlouise — APPROVED** [review]
Looks fantastic! 💯

**Review by alexandec — APPROVED** [review]

**Pull request merged**